src/reducers/contacts.js

```diff
@@ -1,23 +1,41 @@
 import {
-    REFRESH_CONTACTS
+    REFRESH_CONTACTS,
+    SET_CONTACT_SEARCH_RESULTS,
+    SET_CURRENT_SEARCH_QUERY,
 } from '../constants'
 
-const defaultState = []
+const defaultState = {
+    // array of all contacts, unfiltered
+    all: [],
+    // array of filtered contacts based on user search
+    searchResults: [],
+    searchQuery: '',
```

Comment:
Do we also need a `chosen` prop? #question

Reply:
okay so this one is tough. I'd like to just reuse the currently selected contact that we're already storing in state, but we don't have it's UID. I'll circle back on this one